src/frontend/dash/index.ts

```diff
@@ -14,31 +14,41 @@ refreshDelphi()
 setInterval(refreshDelphi, 1000 * 60 * 2) // refresh post every 1 minute
 
 function regenCircles(loggedin: LoggedIn) {
-    const desiredRatio = 1; // y / x
-    const ratioError = .1;
-    let tries = 0;
-    let { maxX, maxY, minX, minY } = getBounds()
     let placedCircles: MemberCircle[] = [];
-    do {
-        tries++;
-        const circles = []
-        const now = Date.now()
-        Object.entries(loggedin).forEach(ent => {
-            const member = members.find(o => o.name == ent[0])
-            circles.push(new MemberCircle(
-                (now - ent[1]) / 1000 / 60 / 60,
-                member.firstname,
-                member.img
-            ))
-        });
-        placedCircles = placeCircles(circles);
 
-        ({ maxX, maxY, minX, minY } = getBounds());
+    const circles = []
+    const now = Date.now()
+    for(const ent in loggedin) {
+        const member = members.find(o => o.name == ent)
+        circles.push(new MemberCircle(
+            (now - loggedin[ent]) / 360000, // 1000 / 60 / 60
+            member.firstname,
+            member.img
+        ))
+    }
 
-    } while (tries < 1000 && ((maxY - minY) / (maxX - minX) < (desiredRatio - ratioError / desiredRatio) || (maxY - minY) / (maxX - minX) > (desiredRatio + ratioError / desiredRatio))) // test to make sure the dimentions are chill
+    setDelphiVisibility(getNameDensity(circles) < 1)
+
+    setAspectRatio(getRatio());
+    placedCircles = placeCircles(circles);
     redrawCircles(placedCircles)
 }
 
+const densityMultiplier = 0.006;
+// Estimates name density
+function getNameDensity(circles : MemberCircle[]) {
```

Comment:
What is name density actually measuring?

Reply:
It's estimating how crowded the display is. I'm using it so I can hide chief delphi.

Reply:
I got that from the lines above. I guess my question is why the calculation for it is what it is. Those seem like pretty random constants being combined in a pretty random way
```js
nameDensityMultiplier * nameSize / Math.sqrt(circleSizeSumSqr / circleSizeSum / circles.length)
```

Reply:
I think he got the constants through trial and error

Reply:
So the constants and the way they are combined are random, he just played around and found something that seemed to work?

Reply:
Yeah. I think the function does need some changing in the future